core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -83,11 +88,50 @@ sealed abstract class Aggregate extends Data {
     }
   }
 
+  // Returns pairs of all fields, element-level and containers, in a Record and their path names
+  def getRecursiveFields(data: Data, path: String): Seq[(Data, String)] = data match {
```

Comment:
Why are we adding 2 new public methods here? I'm guessing these should be `private[chisel3]`?

Reply:
Made them private